### DISCUSSION TITLE
Most useful background patterns

### DISCUSSION THREAD

**Discussion by letsfindaway** [OpenBoard-org/OpenBoard#1169] [Ideas]
Some time ago I have created a PR #888 which enables a flexible and extensible set of background patterns. A description of this feature is also available on a wiki page: [Flexible background patterns](https://github.com/letsfindaway/OpenBoard/wiki/Flexible-background-patterns). Together with this PR I created also an initial set of patterns. This set

- covers the current set of background patterns for OpenBoard,
- adds patterns requested by feature requests,
- and some more which just came to my mind and are useful to demonstrate the capabilities of this PR, e.g. stave lines.

But finally the set of patterns delivered with OpenBoard, and especially those patterns, which are initially visible in the background dialog should be defined based on the needs of the users.

So I'm starting this discussion to

- collect ideas for background patterns to add,
- get input about the most important patterns,
- and finally define the set and sequence of the patterns.

The current list of patterns is:

- blank
- ruled (horizontal lines)
- ruled 5mm
- crossed
- crossed 5mm
- Seyes
- stave
- stave with border
- crossed 1mm
- crossed 5mm with border
- ruled with border
- isometric grid
- German elementary school, 1st grade
- German elementary school, 2nd grade
- German elementary school, 3rd grade

Please add your ideas and needs!

**Comment by ch-westphal** [comment]
Wouldn't it be nice to be able to simply define custom backgrounds by dropping a file (in a format documented in the documentation) into an OpendBoard folder? 

As a personal example, I use three sections in my students' workbooks: lessons, exercises and experiments. I'd like to be able to have three backgrounds with a discreet but explicit ‘lessons’, ‘exercises’ and ‘experiments’ header.

**Reply by letsfindaway** [reply]
> Wouldn't it be nice to be able to simply define custom backgrounds by dropping a file (in a format documented in the documentation) into an OpenBoard folder?

In the end it is like that. There will be a system folder with the patterns delivered with OpenBoard and a user folder for custom patterns. The format is described in the wiki page linked above. Note that the pattern description is finally also part of the document. To deliver a custom pattern, the author of the pattern just has to deliver a document where one of the pages uses that pattern. It is then automatically installed in the recipient's user folder.
 
> As a personal example, I use three sections in my students' workbooks: lessons, exercises and experiments. I'd like to be able to have three backgrounds with a discreet but explicit ‘lessons’, ‘exercises’ and ‘experiments’ header.

Note that the background patterns are only meant to describe (mostly repeating) lines, not any texts, images or so. If you need some kind of form, then a PDF background is the better way,

**Comment by ch-westphal** [comment]
Thanks, I'll have a look

**Comment by bruening-bw** [comment]
I keep it simple and stupid. I use a simple crossed background. But I have a small white margin because it looks better in PDF exports. I created a PNG file (1920x1080) with a line spacing that can also be set in OpenBoard to make the ruler fit to my lines, I add the entry DefaultPageSize=@Size(1920 1080) in the [Board] section of OpenBoardUser.config.

![bck](https://github.com/user-attachments/assets/e9fc53a5-aa8d-41ef-b35a-461add3260d3)

Unfortunately, I have to set the background manually each time and adjust the grid size. Your flexible background patterns would make my life much easier.

**Comment by letsfindaway** [comment]
This feature will be included in OpenBoard 1.8.0. So I'm trying to reactivate this discussion with my initial intent:

> So I'm starting this discussion to
>
> * collect ideas for background patterns to add,
> * get input about the most important patterns,
> * and finally define the set and sequence of the patterns.
